### PR TITLE
Added the swagger docs for PATCH users/self route

### LINKED ISSUE #77: Addition of the Swagger docs for PATCH users/self route.
Add the swagger docs for the PATCH users/self route

### PR DESCRIPTION
**Pull request by Rucha1499**
Added the swagger docs for PATCH users/self route. Fixes #77

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -5,6 +5,45 @@ const usersController = require('../controllers/usersController')
 const userValidator = require('../middlewares/validators/user')
 
 router.post('/', authenticate, userValidator.createUser, usersController.addNewUser)
+
+/**
+ * @swagger
+ * /users/self:
+ *   patch:
+ *     summary: Use to update the user data.
```
Comment (whyDontI) [inline routes/users.js:13]: Please add `requestBody` after this line.
Refer here https://swagger.io/docs/specification/basic-structure/

Comment (Rucha1499) [reply routes/users.js:13]: Done with requested changes. Thank you!

File: utils/swaggerDefinition.js [hunk 2]
```diff
@@ -109,7 +112,7 @@ const swaggerOptions = {
           }
         },
         errors: {
-          unAuthorized: {
+          unauthorized: {
```
Comment (whyDontI) [inline utils/swaggerDefinition.js:115]: Make it `unAuthorized`, as it is referenced in `auth.js` and `healthcheck.js` too, the reference might fail there.

File: routes/users.js [hunk 3]
```diff
@@ -25,11 +72,11 @@ router.get('/', authenticate, usersController.getUsers)
  *             schema:
  *               $ref: '#/components/schemas/users'
  *       401:
- *         description: Unauthorized
+ *         description: unauthorized
  *         content:
  *           application/json:
  *             schema:
- *               $ref: '#/components/schemas/errors/unAuthorized'
+ *               $ref: '#/components/schemas/errors/unauthorized'
```
Comment (whyDontI) [inline routes/users.js:79]: NIT: Revert this line, keep the `unAuthorized` in camel case.

**Review by whyDontI — CHANGES_REQUESTED** [review]

**Review by whyDontI — COMMENTED** [review]
Thanks @Rucha1499 , Awesome work 🔥 👍🏽

File: routes/users.js (hunk 3 above)
Comment (Rucha1499) [reply routes/users.js:79]: okay sure!

**Comment by Rucha1499** [comment]
> Thanks @Rucha1499 , Awesome work 🔥 👍🏽

File: routes/users.js (hunk 3 above)
Comment (Rucha1499) [reply routes/users.js]: Done with changes!

**Review by whyDontI — APPROVED** [review]

**Review by ankurnarkhede — COMMENTED** [review]
LGTM

**Pull request merged**